nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -1583,6 +1626,274 @@ impl DataStore {
     }
 }
 
+pub struct VolumeReplacementParams {
+    pub volume_id: Uuid,
+    pub region_id: Uuid,
+    pub region_addr: SocketAddrV6,
+}
+
+impl DataStore {
+    /// Replace a read-write region in a Volume with a new region.
+    pub async fn volume_replace_region(
+        &self,
+        target: VolumeReplacementParams,
+        replacement: VolumeReplacementParams,
+    ) -> Result<(), Error> {
+        // In a single transaction:
+        //
+        // - set the target region's volume id to the replacement's volume id
```

Comment:
This may be my ignorance about the DB or replacement in general, but why does the replacement have it's own volume id at all, and why is this a step?

Reply:
I wanted to reuse the volume delete saga machinery that will properly delete the old region after the replacement has completed (note this will be problematic in the case of expungement until #5789 lands, but has been useful when testing with manually requested region replacements). For that, a synthetic volume is required, one that only contains the old region to be deleted. Swapping is also required to make the replacement region's volume id correct (as it is written in to the VCR as part of this transaction).